languages/python/polar/query.py

```diff
@@ -40,15 +30,16 @@ def __init__(self, ffi_query, *, host=None):
 
     def __del__(self):
         del self.host
-        lib.query_free(self.ffi_query)
+        del self.ffi_query
 
     def run(self):
         """Run the event loop and yield results."""
         assert self.ffi_query, "no query to run"
         while True:
-            event_s = lib.polar_next_query_event(self.ffi_query)
-            event = ffi_deserialize(event_s)
+            ffi_event = self.ffi_query.next_event()
+            event = json.loads(ffi_event.get())
             if event == "Done":
+                del ffi_event
```

Comment:
```suggestion
            ffi_event = self.ffi_query.next_event()
            event = json.loads(ffi_event.get())
            del ffi_event
            if event == "Done":
```

Reply:
oh yeah you're totally right, I was thinking we still need it while the event loop is running but we don't